### PR TITLE
Fix forge tests post V3

### PR DESCRIPTION
**Pull request by aroralanuk**
### Description

- fixes GasRouter expectRevert message
- added TestMerkleRootHook for proof() and fixes MerkleRootMultisig test
- fixes ERC5164 tests post v3 changes

### Drive-by changes

- added contract name to mailbox and abstractHook error messages
- removed unnecessary tests for "message too large"
- downgrade slither in actions to 0.3.0 because of their recent "missing inheritance" bug on main

### Related issues

- V3

### Backward compatibility

Yes

### Testing

Unit tests

### CODE REVIEW

**Review by yorhodes — APPROVED** [review]

File: .github/workflows/node.yml [hunk 1]
```diff
@@ -174,7 +174,7 @@ jobs:
         run: yarn workspace @hyperlane-xyz/core run test
 
       - name: Run Slither
-        uses: crytic/slither-action@main
+        uses: crytic/slither-action@v0.3.0
```
Comment (yorhodes) [inline .github/workflows/node.yml:177]: call this out in desc

File: solidity/test/isms/ERC5164ISM.t.sol [hunk 2]
```diff
@@ -144,7 +174,7 @@ contract ERC5164ISMTest is Test {
         vm.startPrank(address(executor));
 
         ism.verifyMessageId(messageId);
-        assertTrue(ism.verifiedMessages(messageId));
+        assertTrue(ism.verifiedMessages(messageId).isBitSet(255));
```
Comment (yorhodes) [inline solidity/test/isms/ERC5164ISM.t.sol:177]: hmm
is there a simpler boolean interface for external consumption?

File: solidity/test/lib/mailboxes.ts [hunk 3]
```diff
@@ -89,15 +96,16 @@ export async function dispatchMessageAndReturnMetadata(
 ): Promise<MessageAndMetadata> {
   // Checkpoint indices are 0 indexed, so we pull the count before
   // we dispatch the message.
```
Comment (yorhodes) [inline solidity/test/lib/mailboxes.ts:98]: all this checkpoint stuff is kinda irrelevant to the mailbox now

File: solidity/test/mailbox.test.ts [hunk 4]
```diff
@@ -69,22 +70,13 @@ describe('Mailbox', async () => {
       ));
     });
 
-    it('Does not dispatch too large messages', async () => {
```
Comment (yorhodes) [inline solidity/test/mailbox.test.ts:72]: call this out in desc and link the issue pls

**Review by yorhodes — APPROVED** [review]

File: solidity/test/lib/mailboxes.ts (hunk 3 above)
Comment (aroralanuk) [reply solidity/test/lib/mailboxes.ts:98]: got it

**Pull request merged**